hr_attendance_report_theoretical_time/models/resource_calendar.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 Janik von Rotz <janik.vonrotz@mint-system.ch>
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+from odoo.addons.resource.models.resource_resource import Intervals
+
+
+class ResourceCalendar(models.Model):
+    _inherit = "resource.calendar"
+
+    def _attendance_intervals_batch_exclude_public_holidays(
+        self, start_dt, end_dt, intervals, resources, tz
+    ):
+        partner_id = self.env.context.get("partner_id", False)
+        if not partner_id:
+            employee_id = self.env.context.get("employee_id", False)
+            if employee_id:
+                employee = self.env["hr.employee"].browse(employee_id)
+                partner_id = employee.address_id.id
+            list_by_dates = (
+                self.env["hr.holidays.public"]
+                .get_holidays_list(
+                    start_dt=start_dt.date(),
+                    end_dt=end_dt.date(),
+                    partner_id=partner_id,
+                )
+                .mapped("date")
+            )
+            for resource in resources:
+                interval_resource = intervals[resource.id]
+                attendances = []
+                for attendance in interval_resource._items:
+                    if attendance[0].date() not in list_by_dates:
+                        attendances.append(attendance)
+                intervals[resource.id] = Intervals(attendances)
+            return intervals
```

Comment:
```suggestion
        employee_id = self.env.context.get("employee_id", False)
        if not partner_id and employee_id:
            employee = self.env["hr.employee"].browse(employee_id)
            self = self.with_context(partner_id=employee.address_id.id)
```
It is not necessary to repeat the same code as `hr_holidays_public`, just add the `partner_id` value in the context for the super to do it.

Reply:
hi @victoralmau, changed as suggested and it works great
thanks